node-cli/index.js

```diff
@@ -0,0 +1,70 @@
+const Realm = require("realm");
+const inquirer = require("inquirer");
+const users = require("./users");
+const schemas = require("./schemas");
+const output = require("./output");
+
+/*  Change the logLevel to increase or decrease the 
+    amount of messages you see in the console.
+    Valid options are:
+    fatal, error, warn, info, detail, debug, and trace
+*/
+Realm.Sync.setLogLevel("error");
+
+let realm;
+async function openRealm() {
+  const config = {
+    schema: [schemas.TaskSchema, schemas.UserSchema, schemas.ProjectSchema],
+    sync: {
+      user: users.getAuthedUser(),
+      partitionValue: "myPartition",
+    },
+  };
+  realm = await Realm.open(config);
+}
+
+output.intro();
+
+async function run() {
+  output.header("*** WELCOME ***");
+  output.header(
+    "Please log in to your Realm account or register as a new user."
+  );
+
+  let choice = await inquirer.prompt([
+    {
+      type: "rawlist",
+      name: "start",
+      message: "What do you want to do?",
+      choices: ["Log in", "Register as a new user"],
+    },
+  ]);
+
+  if (choice.start === "Log in") {
+    users.logIn();
+  } else {
+    users.registerUser();
+  }
+}
+
+run().catch((err) => {
+  output.error(err.message);
+});
+
+async function getRealm() {
+  if (realm == undefined) {
+    await openRealm();
+  }
+  return realm;
+}
+
+async function closeRealm() {
```

Comment:
`Realm.close()` is not an async method so `closeRealm` doesn't have to be async.